modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java

```diff
@@ -0,0 +1,145 @@
+import com.liferay.petra.string.StringUtil;
+import com.liferay.portal.kernel.cache.MultiVMPool;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.template.TemplateConstants;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import org.junit.Before;
+
+/**
+ * @author Albert Gomes Cabral
+ */
+public abstract class BaseTemplateUpgradeProcessTestCase {
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+	}
+
+	protected DDMTemplate addDDMTemplate(String pathFile) throws Exception {
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext();
+
+		long classPK = ParamUtil.getLong(serviceContext, "classPK");
+
+		return _ddmTemplateService.addTemplate(
+			_group.getGroupId(), _portal.getClassNameId(DDMTemplate.class),
+			classPK, _portal.getClassNameId(JournalArticle.class),
+			HashMapBuilder.put(
+				LocaleUtil.US, "DDMTemplate Name"
+			).build(),
+			HashMapBuilder.put(
+				LocaleUtil.US, "DDMTemplate Description"
+			).build(),
+			DDMTemplateConstants.TEMPLATE_TYPE_DISPLAY, StringPool.BLANK,
+			TemplateConstants.LANG_TYPE_FTL, read(pathFile), serviceContext);
+	}
+
+	protected FragmentEntry addFragmentEntry(String pathFile) throws Exception {
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(
+				_group.getGroupId(), TestPropsValues.getUserId());
+
+		FragmentCollection fragmentCollection =
+			_fragmentCollectionService.addFragmentCollection(
+				_group.getGroupId(), "Fragment Collection", StringPool.BLANK,
+				serviceContext);
+
+		return _fragmentEntryService.addFragmentEntry(
+			_group.getGroupId(), fragmentCollection.getFragmentCollectionId(),
+			null, "FragmentEntry Name", null, read(pathFile), null, false, null,
+			null, 0, FragmentConstants.TYPE_COMPONENT, null,
+			WorkflowConstants.STATUS_DRAFT, serviceContext);
+	}
+
+	protected String getBasePath() {
+		return "com/liferay/dynamic/data/mapping/dependencies/upgrade";
+	}
+
+	protected abstract String getUpgradeStepClassName() throws Exception;
+
+	protected String read(String pathFile) throws Exception {
```

Comment:
@alberttgomes, please use `private` instead and rename the parameter to `filePath`.